tests/test_train.py

```diff
@@ -358,6 +363,258 @@ def test_train_nlu_autoconfig(
     assert args[1] == autoconfig.TrainingType.NLU
 
 
+def mock_async(monkeypatch, target, name):
+    mock = Mock()
+
+    async def mock_async_func(*args: Any, **kwargs: Any) -> None:
+        mock(*args, **kwargs)
+
+    monkeypatch.setattr(target, name, mock_async_func)
+    return mock
+
+
+def mock_core_training(monkeypatch):
```

Comment:
types

Reply:
added, except for the `target` on line 366, do you know what is the type for it?

Reply:
@joejuzl could you please take this over

Reply:
I can have a look at these in a sec as Joe is having German class as far as I know 👍🏻